### PR TITLE
New package services

### PR DESCRIPTION
**Pull request by msaug**
# HEX-XX Name of your change

Task number **# [(HEX-)](https://hexanax.atlassian.net/browse/HEX-)**

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Testing the changes**:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by bdeleglise — APPROVED** [review]

File: src/main/java/fr/insalyon/pldagile/services/CityMapUtils.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
+package fr.insalyon.pldagile.services;
```
Comment (Lezzio) [inline src/main/java/fr/insalyon/pldagile/services/CityMapUtils.java:1]: Util classes usually provide static methods

File: src/main/java/fr/insalyon/pldagile/services/CityMapUtils.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+ * Contains methods that allow us to compute the center of a CityMap,
+ * the coordinates of a x <a href = https://en.wikipedia.org/wiki/Minimum_bounding_rectangle#:~:text=The%20minimum%20bounding%20rectangle%20(MBR,)%2C%20max(y).>Minimum Bounding Rectangle</a>,
+ * and the optimal zoom value to have a full view of the CityMap.
+ */
+public class CityMapUtils {
+
+    private Map<Long, Intersection> intersections;
+
+
+    public CityMapUtils(CityMap cityMap) {
```
Comment (Lezzio) [inline src/main/java/fr/insalyon/pldagile/services/CityMapUtils.java:20]: Empty constructor and put the CityMap as a parameter in each function

File: src/test/java/fr/insalyon/pldagile/tsp/SimulatedAnnealingTest.java [hunk 3]
```diff
@@ -64,7 +66,7 @@ void runSimulatedAnnealing() {
     }
 
     @Test
-    void getTotalDistance() {
+    void getTotalDistance() throws InterruptedException {
```
Comment (Lezzio) [inline src/test/java/fr/insalyon/pldagile/tsp/SimulatedAnnealingTest.java:69]: We should test if when the exception is thrown it invalidates the test or if we should catch it instead

**Review by Lezzio — CHANGES_REQUESTED** [review]

File: src/main/java/fr/insalyon/pldagile/services/CityMapUtils.java (hunk 1 above)
Comment (msaug) [reply src/main/java/fr/insalyon/pldagile/services/CityMapUtils.java:1]: done

File: src/test/java/fr/insalyon/pldagile/tsp/SimulatedAnnealingTest.java (hunk 3 above)
Comment (msaug) [reply src/test/java/fr/insalyon/pldagile/tsp/SimulatedAnnealingTest.java:69]: there's actually no InterruptedException

**Pull request merged**